lib/primo/parameter_validatable.rb

```diff
@@ -23,20 +23,26 @@ module ParameterValidatable
     #######################################33
 
 
-
     def validate(params)
       validators.each do |validate|
         message = validate[:message][params]
         if !send(validate[:query], params)
-          raise error.new(message)
+          error = error_class.new(message)
+          if error.respond_to?(:endpoint_url) && params.respond_to?(:request) && params.request.respond_to?(:uri)
+            error.endpoint_url = params.request.uri.to_s
+          end
+          if error.respond_to?(:endpoint_status) && params.respond_to?(:response) && params.response.respond_to?(:code)
+            error.endpoint_status = params.response.code
+          end
+          raise error
```

Comment:
@quoideneuf hmm this looks promising, but couldn't we just  raise `error_class.new(params)` and avoid the extra logic? (note that `StandardError.new`  is perfectly happy to take a hash as a param which will be converted to string).

I would prefer that we just pass the params, but in lieu of that I would prefer that we move the logic into the error function (have it return what we want from there).

Reply:
@dkinzer I think the goal of getting the endpoint into honeybadger can be achieved by just appending to the endpoint. So I've taken out all the additional methods and just done that.